src/deadline/client/cli/_groups/manifest_group.py

```diff
@@ -0,0 +1,237 @@
+
+
+@click.group(name="manifest")
+@_handle_error
+def cli_manifest():
+    """
+    Commands to work with AWS Deadline Cloud Job Attachments.
+    """
+
+
+@cli_manifest.command(
+    name="snapshot",
+    help="BETA - Generates a snapshot of files in a directory root as a Job Attachment Manifest.",
+)
+@click.option("--root", required=True, help="The root directory to snapshot. ")
+@click.option(
+    "--destination",
+    default=None,
+    help="Destination directory where manifest is created. Defaults to the manifest root directory.",
+)
+@click.option(
+    "--name",
+    default=None,
+    help="Name of the manifest. A timestamp is added YYYY-MM-DD-HH-MM-SS for versioning.",
+)
+@click.option(
+    "--glob",
+    default=None,
+    help="Glob include and exclude of directory and file regex to include in the manifest.",
+)
+@click.option("--diff", default=None, help="File Path to Asset Manifest to diff against.")
+@click.option("--json", default=None, is_flag=True, help="Output is printed as JSON for scripting")
+@_handle_error
+def manifest_snapshot(
+    root: str, destination: str, name: str, glob: str, diff: str, json: bool, **args
+):
+    """
+    Creates manifest of files specified by root directory.
+    """
+    logger: ClickLogger = ClickLogger(is_json=json)
+    if not os.path.isdir(root):
+        raise NonValidInputError(f"Specified root directory {root} does not exist. ")
+
+    if destination and not os.path.isdir(destination):
+        raise NonValidInputError(f"Specified destination directory {destination} does not exist. ")
+    elif destination is None:
+        destination = root
+        logger.echo(f"Manifest creation path defaulted to {root} \n")
+
+    manifest_out = _manifest_snapshot(
+        root=root, destination=destination, name=name, glob=glob, diff=diff, logger=logger
+    )
+    if manifest_out:
+        logger.json(dataclasses.asdict(manifest_out))
+
+
+@cli_manifest.command(
+    name="diff", help="BETA - Compute a directory root diff of new, modified or deleted files."
+)
+@click.option("--root", help="The root directory to compare changes to. ")
+@click.option(
+    "--manifest",
+    required=True,
+    help="The path to manifest file to diff against.",
+)
+@click.option(
+    "--glob",
+    default=None,
+    help="Glob regex to find include and exclude directories and files to include in the manifest.",
+)
+@click.option("--json", default=None, is_flag=True, help="Output is printed as JSON for scripting")
+@_handle_error
+def manifest_diff(root: str, manifest: str, glob: str, json: bool, **args):
+    """
+    Check file differences between a directory and specified manifest.
+    """
+    logger: ClickLogger = ClickLogger(is_json=json)
+    if not os.path.isfile(manifest):
+        raise NonValidInputError(f"Specified manifest file {manifest} does not exist. ")
+
+    if not os.path.isdir(root):
+        raise NonValidInputError(f"Specified root directory {root} does not exist. ")
```

Comment:
nit - the error can be misleading, it doesn't apply to path exist but wrong types?